src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.cs

```diff
@@ -491,6 +542,124 @@ private int FindEndPosition(ReadOnlySpan<char> input, int i)
             return false;
         }
 
+        /// <summary>Find match end position using the original pattern, end position is known to exist. This version also produces captures.</summary>
+        /// <param name="input">input span</param>
+        /// <param name="i">inclusive start position</param>
+        /// <param name="resultRegisters">out parameter for the final register values, which indicate capture starts and ends</param>
+        /// <returns></returns>
+        private int FindEndPositionCapturing(ReadOnlySpan<char> input, int i, out Registers resultRegisters)
+        {
+            int i_end = input.Length;
+            Registers endRegisters = default(Registers);
+
+            // Pick the correct start state based on previous character kind.
+            uint prevCharKind = GetCharKind(input, i - 1);
+            DfaMatchingState<TSetType> state = _initialStates[prevCharKind];
+            // Initialize registers with -1, which means "not seen yet"
+            Registers initialRegisters = new Registers
+            {
+                captureStarts = new int[_capsize],
+                captureEnds = new int[_capsize],
+            };
```

Comment:
At this point we're 100% sure there's a match and we just need to get the details, right?  Just confirming we're not potentially allocating these unnecessarily.